util/logtools/web_splitter.ts

```diff
@@ -301,13 +303,51 @@ const doExport = (state: PageState): void => {
   window.URL.revokeObjectURL(a.href);
 };
 
+const doExportAll = (state: PageState): void => {
+  const anonymizeLogs = state.anonInput.checked;
+  const analysisFilter = state.analysisFilterInput.checked;
+  const output: string[] = [];
+  const notifier = new WebNotifier(state.errorDiv);
+  const anonymizer = new Anonymizer();
+  const splitter = new Splitter('', '', notifier, true, analysisFilter);
+
+  for (const line of state.lines) {
+    splitter.processWithCallback(line, true, (line) => {
+      if (anonymizeLogs) {
+        const anonLine = anonymizer.process(line, notifier);
+        if (anonLine === undefined)
+          return;
+        output.push(anonLine);
+      } else {
+        output.push(line);
+      }
+    });
+  }
+
+  if (anonymizeLogs) {
+    anonymizer.validateIds(notifier);
+    for (const line of output)
+      anonymizer.validateLine(line, notifier);
+  }
+
+  const filename = 'processed.log';
+  const blob = new Blob([output.join('\n')], { type: 'text/plain' });
+  const a = document.createElement('a');
+  a.setAttribute('download', filename);
+  a.href = window.URL.createObjectURL(blob);
+  a.click();
+  window.URL.revokeObjectURL(a.href);
```

Comment:
I'm likely missing something obvious here, but could we hoist this out rather than duplicating it? Off the top of my head the validation and file generation look the same between `exportAll` and `exportSelected`

Reply:
Yeah, this can definitely be hoisted out.

I debated for a while whether to just put all the new `doExportAll` stuff inside the (then-`doExport`, now `doExportSelected`), and just have a single func with a bool param to indicate which.  But it would have involved so many conditionals in the main func that it would have been pretty fugly code, and new, shorter func seemed simpler.  But this is a good improvement to further reduce some of the duplication.

I'll add it before merging (but likely won't be until tomorrow - new gaming PC is finished, so time to go play with the new toy for a bit 🙃 ).

Reply:
Yeah, I definitely agree with the two different functions overall, it was mostly a question of "these two specific segments look identical". Good luck with the new PC!